index.bs

```diff
@@ -16,13 +16,439 @@ WPT Display: closed
+</div>
+
+Issue: The Fetch specification does not yet provide an interface through which other specifications can interact with the cookie store, so it is not yet possible to express the above algorithm with the appropriate level of rigor. Once the cookie store has a well-defined interface that is usable from other specifications, this algorithm can make use of it.
+
+
+Storage-Access Headers {#headers}
+=================================
+
+The following sections define a request header and a response header. The request header exposes information about the [=request=]'s access to cookies to a server. The response header allows a server to opt into accessing cross-site cookies on a particular request or when loading an <{iframe}>. 
+
+The `Sec-Fetch-Storage-Access` HTTP Request Header {#sec-fetch-storage-access-header}
+-----------------------------------------------------------------
+
+The <dfn http-header export>`Sec-Fetch-Storage-Access`</dfn> HTTP request header exposes a [=request=]'s
+ability or inability to access cookies to a server. It is a [=Structured Field=] [=structured field/item=] whose value MUST be a
+[=structured field/token=]. [[!I-D.structured-field-values-for-http]] Its ABNF is:
+
+```
+Sec-Fetch-Storage-Access = sf-token
+```
+
+Valid `Sec-Fetch-Storage-Access` values include "<code>[=storage access status/none=]</code>", "<code>[=storage access status/inactive=]</code>", and
+"<code>[=storage access status/active=]</code>". In order to support forward-compatibility with as-yet-unknown
+semantics, servers SHOULD ignore this header if it contains an invalid value.
+
+<pre class="example" id="sec-fetch-storage-access-usage">
+
+// When the request's credentials mode is "`omit`", the header is omitted:
+
+
+// When the request is same-site, the header is omitted:
+
+
+// When the request has no access to cross-site cookies, the header is "<code>[=storage access status/none=]</code>":
+Sec-Fetch-Storage-Access: none
+
+// When the request has no access to cross-site cookies, but
+// 'storage-access' permission has already been granted, the header is
+// "<code>[=storage access status/inactive=]</code>":
+Sec-Fetch-Storage-Access: inactive
+
+// When the request has access to cross-site cookies, the header is "<code>[=storage access status/active=]</code>":
+Sec-Fetch-Storage-Access: active
+
+</pre>
+
+<div algorithm>
+    To <dfn export lt="set-storage-access">set the `Sec-Fetch-Storage-Access` header</dfn> for a [=request=] |request|:
+
+    <ol class="algorithm">
+        1.  Assert: |request|'s [=request/url=] is a [=potentially trustworthy URL=].
+        1. If |request|'s [=request/credentials mode=] [=string/is=] not "`include`", abort these steps.
+        1.  Let |access| be |request|'s [=request/storage access status=].
+        1.  If |access| is null, abort these steps.
+        1.  Let |value| be a [=Structured Field=] value whose value is a [=structured field/token=].
```

Comment:
nit: I wonder if we should reference "structured field value" as https://fetch.spec.whatwg.org/#concept-header-list-set-structured-header does?